pkg/operator/controller/prometheus.go

```diff
@@ -134,166 +125,15 @@ func isPrometheusDeployed(logger logr.Logger, c client.Client, namespace string)
-				componentAlertLabelKey:    componentAlertLabelValue,
-			},
-		),
-		generateAlertRule(
-			"CDINoDefaultStorageClass",
-			`sum(kubevirt_cdi_storageprofile_info{default="true"} or on() vector(0)) +
-			sum(kubevirt_cdi_storageprofile_info{virtdefault="true"} or on() vector(0)) +
-			(count(kubevirt_cdi_datavolume_pending == 0) or on() vector(0)) == 0`,
-			promv1.Duration("5m"),
-			map[string]string{
-				"summary":     "No default StorageClass or virtualization StorageClass, and a DataVolume is pending for one",
-				"runbook_url": fmt.Sprintf(runbookURLTemplate, "CDINoDefaultStorageClass"),
-			},
-			map[string]string{
-				severityAlertLabelKey:     "warning",
-				healthImpactAlertLabelKey: "none",
-				partOfAlertLabelKey:       partOfAlertLabelValue,
-				componentAlertLabelKey:    componentAlertLabelValue,
-			},
-		),
-		generateAlertRule(
-			"CDIMultipleDefaultVirtStorageClasses",
-			`sum(kubevirt_cdi_storageprofile_info{virtdefault="true"} or on() vector(0)) > 1`,
-			promv1.Duration("5m"),
-			map[string]string{
-				"summary":     "More than one default virtualization StorageClass detected",
-				"runbook_url": fmt.Sprintf(runbookURLTemplate, "CDIMultipleDefaultVirtStorageClasses"),
-			},
-			map[string]string{
-				severityAlertLabelKey:     "warning",
-				healthImpactAlertLabelKey: "none",
-				partOfAlertLabelKey:       partOfAlertLabelValue,
-				componentAlertLabelKey:    componentAlertLabelValue,
-			},
-		),
-		generateAlertRule(
-			"CDIDefaultStorageClassDegraded",
-			`sum(kubevirt_cdi_storageprofile_info{default="true",rwx="true",smartclone="true"} or on() vector(0)) +
-			sum(kubevirt_cdi_storageprofile_info{virtdefault="true",rwx="true",smartclone="true"} or on() vector(0)) == 0`,
-			promv1.Duration("5m"),
-			map[string]string{
-				"summary":     "Default storage class has no smart clone or ReadWriteMany",
-				"runbook_url": fmt.Sprintf(runbookURLTemplate, "CDIDefaultStorageClassDegraded"),
-			},
-			map[string]string{
-				severityAlertLabelKey:     "warning",
-				healthImpactAlertLabelKey: "none",
-				partOfAlertLabelKey:       partOfAlertLabelValue,
-				componentAlertLabelKey:    componentAlertLabelValue,
-			},
-		),
-	}
-}
-
 func newPrometheusRule(namespace string) *promv1.PrometheusRule {
-	runbookURLTemplate := getRunbookURLTemplate()
+	promRule, err := rules.BuildPrometheusRule(namespace)
+	if err != nil {
+		panic(err)
+	}
 
 	return &promv1.PrometheusRule{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      ruleName,
-			Namespace: namespace,
-			Labels: map[string]string{
-				common.CDIComponentLabel:  "",
-				common.PrometheusLabelKey: common.PrometheusLabelValue,
-			},
-		},
-		Spec: promv1.PrometheusRuleSpec{
-			Groups: []promv1.RuleGroup{
-				{
-					Name:  "cdi.rules",
-					Rules: append(getRecordRules(namespace), getAlertRules(runbookURLTemplate)...),
-				},
-			},
-		},
+		ObjectMeta: promRule.ObjectMeta,
+		Spec:       promRule.Spec,
 	}
```

Comment:
this function can be deleted, and its usage replaced by `rules.BuildPrometheusRule(namespace)`, you are creating a new `PrometheusRule` with the contents of another `PrometheusRule`

Reply:
We wrapped it in a func